docs/posts/bigquery-arrays/index.qmd

```diff
@@ -0,0 +1,240 @@
+---
+title: Working with arrays in Google BigQuery
+author: "Phillip Cloud"
+date: "2023-09-12"
+categories:
+    - release
+    - blog
+    - bigquery
+    - arrays
+    - cloud
+---
+
+## Introduction
+
+Ibis and BigQuery have [worked well together for years](https://cloud.google.com/blog/products/data-analytics/ibis-and-bigquery-scalable-analytics-comfort-python).
+
+In Ibis 7.0.0, they work even better together with the addition of array
+functionality for BigQuery.
+
+Let's look at some examples using BigQuery's [IMDB data](https://developer.imdb.com/non-commercial-datasets/).
+
+## Basics
+
+First we'll connect to BigQuery and pluck out a table to work with.
+
+We'll start with `from ibis.interactive import *` for maximum convenience.
+
+```{python}
+from ibis.interactive import *
+
+con = ibis.connect("bigquery://ibis-gbq")  # <1>
+con.set_database("bigquery-public-data.imdb")  # <2>
+```
+
+1. Connect to the **billing** project. Compute (but not storage) is billed to
+   this project.
+2. Set the database to the project and dataset that we will use for analysis.
+
+Let's look at the tables in this dataset:
+
+```{python}
+con.list_tables()
+```
+
+Let's pull out the `name_basics` table, which contains names and metadata about
+people listed on IMDB. We'll call this `ents` (short for `entities`), and remove some
+columns we won't need:
+
+```{python}
+ents = con.tables.name_basics.drop("birth_year", "death_year")
+ents
+```
+
+### Splitting strings into arrays
+
+We can see that `known_for_titles` looks sort like an array, so let's call the [`split()`]()
+method on that column and replace the existing column:
+
+```{python}
+ents = ents.mutate(known_for_titles=_.known_for_titles.split(","))
+ents
+```
+
+Similarly for `primary_profession`, since people involved in show business often
+have more than one responsibility on a project:
+
+```{python}
+ents = ents.mutate(primary_profession=_.primary_profession.split(","))
+```
+
+### Array length
+
+Let's see how many titles each entity is known, and then show the five
+people with the largest number of titles they're known for:
+
+This is computed using the
+[`length()`](../../reference/expression-collections.qmd#ibis.expr.types.arrays.ArrayValue.length)
+API on array expressions:
+
+```{python}
+(
+    ents.select("primary_name", num_titles=_.known_for_titles.length())
+    .order_by(_.num_titles.desc())
+    .limit(5)
+)
+```
+
+It seems like the length of the `known_for_titles` might be capped at five!
+
+### Index
+
+We can see the position of `"actor"` in `primary_profession`s:
+
+```{python}
+ents.primary_profession.index("actor")
+```
+
+A return value of `-1` indicates that `"actor"` is not present in the value:
+
+Let's check whether `"actor"` shows up in a different position in the `primary_profession` column:
```

Comment:
This sentence threw me for a second (although I'm not looking at the rendered output, either).

Maybe a touch more explicit prose re: looking for entities who aren't primarily actors?

Reply:
Reworded